File: Kotlin/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.kt
class Solution {
    fun solution(s: String, skip: String, index: Int): String {

        val sChar = s.toCharArray()
        val skipSet = skip.toSet()

        val sb = StringBuilder()

        for (i in sChar.indices) {
            var char = sChar[i] //y -> a

            repeat(index) {
                do {
                    char++
                    println(char)
                    if (char > 'z') char -= 26
                } while (char in skipSet)
            }

            sb.append(char)
        }

        return sb.toString()
    }
}

File: C#/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cs
using System;
using System.Text;

public class Solution {
    public string solution(string s, string skip, int index) {
        
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < s.Length; i++)
        {
            char c = s[i];
            for(int j = 0; j < index; j++)
            {
                c++;
                if('z' < c) c = 'a';
                
                for(int k = 0; k < skip.Length; k++)
                {
                    if(skip.Contains(c)) c++;
                    if('z' < c) c = 'a';
                }
            }
            
            sb.Append(c);
        }
        
        return sb.ToString();
    }
}